core/trino-main/src/main/java/io/trino/connector/system/RuleStatsSystemTable.java

```diff
@@ -64,7 +64,7 @@ public RuleStatsSystemTable(Optional<RuleStatsRecorder> ruleStatsRecorder, TypeM
                 .column("matches", BIGINT)
                 .column("failures", BIGINT)
                 .column("average_time", DOUBLE)
-                .column("time_distribution_percentiles", typeManager.getType(mapType(DOUBLE.getTypeSignature(), DOUBLE.getTypeSignature())))
+                .column("time_distribution_percentiles", new MapType(DOUBLE, DOUBLE, typeManager.getTypeOperators()))
```

Comment:
The previous way of constructing the map abstracted away the map type dependencies (the key hash/equal/identical operators). This was a good thing.

The caller of `new MapType` needs to provide type operators. if they provide `new TypeOperators()`, things will still work, but likely not as well as they should.
Should the `MapType` constructor take `TypeManager` to nudge toward correct use of the API?

Reply:
Also, the `Create Array, Map, and Row types directly` commit doesn't have any justification, so it's unclear what's the intention here. 

- Is `TypeManager.getType(TypeSignature)` supposed to be deprecated?
- add a line of explanation to the commit message

Reply:
I know. This is one of those unfortunate inconsistencies we have across the codebase. Except for MapType, every other type can be constructed directly, and in most cases (but not all) are. Even MapType is constructed directly in a large number of callsites, with TypeOperators passed explicitly.

This is an attempt at cleaning up usages and bringing some uniformity. This will evolve as I continue to refactor how types, constraints and type inference are handled.

> Is TypeManager.getType(TypeSignature) supposed to be deprecated?

Not yet, but it will take a different shape once I revamp TypeSignature.